command/v7/labels_command_test.go

```diff
@@ -46,117 +46,215 @@ var _ = Describe("labels command", func() {
-		})
+				It("returns the error", func() {
+					Expect(executeErr).To(MatchError("boom"))
+				})
 
-		It("retrieves the labels associated with the application", func() {
-			Expect(fakeActor.GetApplicationByNameAndSpaceCallCount()).To(Equal(1))
-			appName, spaceGUID := fakeActor.GetApplicationByNameAndSpaceArgsForCall(0)
-			Expect(appName).To(Equal("dora"))
-			Expect(spaceGUID).To(Equal("some-space-guid"))
-		})
+				It("still prints all warnings", func() {
+					Expect(testUI.Err).To(Say("some-warning-1"))
+					Expect(testUI.Err).To(Say("some-warning-2"))
+				})
 
-		It("displays the labels that are associated with the application, alphabetically", func() {
-			Expect(testUI.Out).To(Say(`Key\s+Value`))
-			Expect(testUI.Out).To(Say(`some-label\s+some-value`))
-			Expect(testUI.Out).To(Say(`some-other-label\s+some-other-value`))
+				It("doesn't say ok", func() {
+					Expect(testUI.Out).ToNot(Say("OK"))
+				})
+			})
+
+			When("checking targeted org and space fails", func() {
+				BeforeEach(func() {
+					fakeSharedActor.CheckTargetReturns(errors.New("nope"))
+				})
+
+				It("returns an error", func() {
+					Expect(executeErr).To(MatchError("nope"))
+				})
+			})
+
+			When("fetching the current user's name fails", func() {
+				BeforeEach(func() {
+					fakeConfig.CurrentUserNameReturns("some-user", errors.New("boom"))
+				})
+
+				It("returns an error", func() {
+					Expect(executeErr).To(MatchError("boom"))
+				})
+			})
 		})
 
-		When("CAPI returns warnings", func() {
+		Describe("for orgs", func() {
 			BeforeEach(func() {
-				fakeActor.GetApplicationByNameAndSpaceReturns(
-					v7action.Application{
-						Metadata: &v7action.Metadata{
-							Labels: map[string]types.NullString{
-								"some-other-label": types.NewNullString("some-other-value"),
-								"some-label":       types.NewNullString("some-value"),
-							},
-						},
+				fakeConfig.CurrentUserNameReturns("some-user", nil)
+				cmd.RequiredArgs = flag.LabelsArgs{
+					ResourceType: "org",
+					ResourceName: "fake-org",
+				}
+				fakeLabelActor.GetOrganizationLabelsReturns(
+					map[string]types.NullString{
+						"some-other-label": types.NewNullString("some-other-value"),
+						"some-label":       types.NewNullString("some-value"),
 					},
-					v7action.Warnings([]string{"some-warning-1", "some-warning-2"}),
+					v7action.Warnings{},
 					nil)
 			})
 
-			It("prints all warnings", func() {
-				Expect(testUI.Err).To(Say("some-warning-1"))
-				Expect(testUI.Err).To(Say("some-warning-2"))
+			It("doesn't error", func() {
+				Expect(executeErr).ToNot(HaveOccurred())
 			})
-		})
 
-		When("there is an error retrieving the application", func() {
-			BeforeEach(func() {
-				fakeActor.GetApplicationByNameAndSpaceReturns(
-					v7action.Application{},
-					v7action.Warnings([]string{"some-warning-1", "some-warning-2"}),
-					errors.New("boom"))
+			It("checks that the user is logged in and targeted to an org and space", func() {
```

Comment:
The example wording is wrong - this is only checking that the user is logged in, not that they are targeted to an org and space.